pkg/backend/openshiftcluster/create.go

```diff
@@ -93,28 +137,21 @@ func (m *Manager) Create(ctx context.Context) error {
 
 	masterZones, err := m.env.Zones(string(m.doc.OpenShiftCluster.Properties.MasterProfile.VMSize))
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if len(masterZones) == 0 {
 		masterZones = []string{""}
 	}
 
 	workerZones, err := m.env.Zones(string(m.doc.OpenShiftCluster.Properties.WorkerProfiles[0].VMSize))
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if len(workerZones) == 0 {
 		masterZones = []string{""}
 	}
 
-	platformCreds := &installconfig.PlatformCreds{
-		Azure: &icazure.Credentials{
-			TenantID:       m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.TenantID,
-			ClientID:       m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientID,
-			ClientSecret:   m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientSecret,
-			SubscriptionID: r.SubscriptionID,
-		},
-	}
+	resourceGroup := m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID[strings.LastIndexByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')+1:]
```

Comment:
This way of getting the resource group name from the resource group ID seems a little unwieldy. Would it be worth adding a function somewhere in the same vein as `ParseResourceID` in `github.com/Azure/go-autorest/autorest/azure/azure.go`? I did write such a function, `ParseResourceGroupID`, but ultimately removed it as it ended up not being needed at the time: https://github.com/Azure/ARO-RP/pull/151#discussion_r376461223

I see this same assignment in both `create.go` and `delete.go`. I guess it might not be worth it if these are the only places it will ever be needed. Just an idea.

Reply:
Here are some other places a new utility function could be used (which I found while looking at https://github.com/Azure/ARO-RP/pull/163):

- https://github.com/Azure/ARO-RP/pull/163/files#diff-90259a58ac8d65a2b486198e4eebb9c3R204
- https://github.com/Azure/ARO-RP/pull/163/files#diff-c5ed0489db136e8ac555a615e08f3676R66
- https://github.com/Azure/ARO-RP/pull/163/files#diff-81e3b7c6dadccc22c9caa1d8c00428d3R12